### PR TITLE
Update link for Announcement Page

### PR DESCRIPTION
**Pull request by kaxil**
https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=6269459 has been archived

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

File: dev/README.md [hunk 1]
```diff
@@ -578,4 +578,5 @@ Cheers,
 </p>
 </details>
 
-- Update the Announce page on [the wiki](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=62694597)
+- Update the Announcement page on
+[Airflow Site](https://github.com/apache/airflow-site/tree/master/landing-pages/site/content/en/announcements)
```
Comment (turbaszek) [inline dev/README.md]: Should be more explicit?
```suggestion
[Airflow Site](https://github.com/apache/airflow-site/tree/master/landing-pages/site/content/en/announcements/_index.md)
```

Comment (potiuk) [reply dev/README.md]: I'd even say - why not link directly to https://airflow.apache.org/announcements/ ? 
When I updated it, I used the "suggest change on this page" link at the bottom and it worked perfectly :)

Comment (kaxil) [reply dev/README.md]: cool will do that

Comment (kaxil) [inline dev/README.md]:
```suggestion
[Airflow Site](https://airflow.apache.org/announcements/ )
```

File: dev/README.md [hunk 2]
```diff
@@ -578,4 +578,5 @@ Cheers,
 </p>
 </details>
 
-- Update the Announce page on [the wiki](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=62694597)
+- Update the Announcement page on
+[Airflow Site](https://airflow.apache.org/announcements/ )
```
Comment (kaxil) [inline dev/README.md]:
```suggestion
[Airflow Site](https://airflow.apache.org/announcements/)
```

**Pull request merged**